# Pandas Assignment


## Part 1

In [256]:
import os
import pandas as pd

##### Read in the movie data: `pd.read_table`

In [257]:
def get_movie_data():
    
    unames = ['user_id','gender','age','occupation','zip']
    users = pd.read_table(os.path.join('../data','users.dat'), 
                          sep='::', header=None, names=unames)
    
    rnames = ['user_id', 'movie_id', 'rating', 'timestamp']
    ratings = pd.read_table(os.path.join('../data', 'ratings.dat'), 
                            sep='::', header=None, names=rnames)
    
    mnames = ['movie_id', 'title','genres']
    movies = pd.read_table(os.path.join('../data', 'movies.dat'), 
                           sep='::', header=None, names=mnames)

    return users, ratings, movies

In [258]:
users, ratings, movies = get_movie_data()

C:\Users\rinty\Anaconda3\lib\site-packages\ipykernel_launcher.py:5: FutureWarning: read_table is deprecated, use read_csv instead.
  """
C:\Users\rinty\Anaconda3\lib\site-packages\ipykernel_launcher.py:5: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.
  """
C:\Users\rinty\Anaconda3\lib\site-packages\ipykernel_launcher.py:9: FutureWarning: read_table is deprecated, use read_csv instead.
  if __name__ == '__main__':
C:\Users\rinty\Anaconda3\lib\site-packages\ipykernel_launcher.py:9: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.
  if __name__ == '__main__':
C:\Users\rinty\Anaconda3\lib\site-packages\ipykernel_l

In [259]:
print (users.head())

   user_id gender  age  occupation    zip
0        1      F    1          10  48067
1        2      M   56          16  70072
2        3      M   25          15  55117
3        4      M   45           7  02460
4        5      M   25          20  55455


In [260]:
print (ratings.head())

   user_id  movie_id  rating  timestamp
0        1      1193       5  978300760
1        1       661       3  978302109
2        1       914       3  978301968
3        1      3408       4  978300275
4        1      2355       5  978824291


In [261]:
print (movies.head())

   movie_id                               title                        genres
0         1                    Toy Story (1995)   Animation|Children's|Comedy
1         2                      Jumanji (1995)  Adventure|Children's|Fantasy
2         3             Grumpier Old Men (1995)                Comedy|Romance
3         4            Waiting to Exhale (1995)                  Comedy|Drama
4         5  Father of the Bride Part II (1995)                        Comedy


##### Clean up the `movies`

- Get the `year`
- Shorten the `title`


In [262]:
tmp = movies.title.str.extract('(.*) \(([0-9]+)\)')
tmp.apply(lambda x:x[0] if len(x) > 0 else None)
tmp.apply(lambda x: x[0][:40] if len(x) > 0 else None)

0    Toy Story
1         1995
dtype: object

In [263]:
movies['year'] = tmp[1]
movies['short_title'] = tmp[0]

In [264]:
print (movies.head())

   movie_id                               title                        genres  \
0         1                    Toy Story (1995)   Animation|Children's|Comedy   
1         2                      Jumanji (1995)  Adventure|Children's|Fantasy   
2         3             Grumpier Old Men (1995)                Comedy|Romance   
3         4            Waiting to Exhale (1995)                  Comedy|Drama   
4         5  Father of the Bride Part II (1995)                        Comedy   

   year                  short_title  
0  1995                    Toy Story  
1  1995                      Jumanji  
2  1995             Grumpier Old Men  
3  1995            Waiting to Exhale  
4  1995  Father of the Bride Part II  


##### Join the tables with `pd.merge` (10 pts)

In [265]:
MergedData = pd.merge(pd.merge(users, ratings, on='user_id'), movies, on='movie_id')
MergedData.head()

user_id gender  age  occupation    zip  movie_id  rating  timestamp  \
0        1      F    1          10  48067      1193       5  978300760   
1        2      M   56          16  70072      1193       5  978298413   
2       12      M   25          12  32793      1193       4  978220179   
3       15      M   25           7  22903      1193       4  978199279   
4       17      M   50           1  95350      1193       5  978158471   

                                    title genres  year  \
0  One Flew Over the Cuckoo's Nest (1975)  Drama  1975   
1  One Flew Over the Cuckoo's Nest (1975)  Drama  1975   
2  One Flew Over the Cuckoo's Nest (1975)  Drama  1975   
3  One Flew Over the Cuckoo's Nest (1975)  Drama  1975   
4  One Flew Over the Cuckoo's Nest (1975)  Drama  1975   

                       short_title  
0  One Flew Over the Cuckoo's Nest  
1  One Flew Over the Cuckoo's Nest  
2  One Flew Over the Cuckoo's Nest  
3  One Flew Over the Cuckoo's Nest  
4  One Flew Over the Cuckoo's Nest

##### What's the highest rated movie? (10 pts))

It counts how many time each movie was watched from 'short_title' column by using value_counts()

In [266]:
MergedData.short_title.value_counts().head()

American Beauty                                   3428
Star Wars: Episode IV - A New Hope                2991
Star Wars: Episode V - The Empire Strikes Back    2990
Star Wars: Episode VI - Return of the Jedi        2883
Jurassic Park                                     2672
Name: short_title, dtype: int64

'GroupbyData' groups the 'short_title' column based on movie names by using groupby and finds the median of rating for each movie from 'rating' column by using median(). Then it sorts the rating column in decending order by using sort_values(ascending=False).

In [267]:
GroupbyData = MergedData.groupby('short_title')['rating'].median()
GroupbyData.sort_values(ascending=False).head()

short_title
Rear Window              5.0
Chinatown                5.0
It's a Wonderful Life    5.0
One Little Indian        5.0
Seven Chances            5.0
Name: rating, dtype: float64

**"American Beauty"** is the highest rated movie. 

**Explanation:** Based on my first analysis, 'American Beauty' was the most watched movie out of all movies. It was watched 3428 times. Then based on my second analysis, median of rating for the movie 'American Beauty' is 5/5. Therefore, American Beauty has the maximum rating and also watched the most. So, it's the hightest rated movie. 

###### What is a good rated movie for date night? (30 pts)

- Hint - highly rated movie by 
    - both partners (might be the same gender or not),
    - based on genre preferences,
    - age group can also be combined

**My partner**
    - Gender: Male
    - Genre preferences: Thriller
    - Age group: 28-35

'MyPartnerData' contains data for age group between (28-35), male gender, and genre is Comedy|Thriller(combination of both partner)

In [268]:
MyPartnerData = MergedData[(MergedData.age >= 28) & (MergedData.age <= 35) & (MergedData.gender == 'M') & (MergedData.genres == 'Comedy|Thriller')]
MyPartnerData.head()

user_id gender  age  occupation    zip  movie_id  rating  timestamp  \
180385      218      M   35          14  95822      3893       5  990465369   
180399      327      M   35          18  55448      3893       1  976397578   
180400      329      M   35           7  02115      3893       1  976475435   
180402      335      M   35          18  55434      3893       1  976362637   
180403      338      M   35           7  55116      3893       4  976348355   

                     title           genres  year  short_title  
180385  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty  
180399  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty  
180400  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty  
180402  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty  
180403  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty

'MyPartnerGroupbyData' groups the 'short_title' column based on movie names by using groupby and finds the median of rating for each movie from 'rating' column by using median(). Then it sorts the rating column in decending order by using sort_values(ascending=False).

In [269]:
MyPartnerGroupbyData = MyPartnerData.groupby('short_title')['rating'].median()
MyPartnerGroupbyData.sort_values(ascending=False)

short_title
Zero Effect             4
Mona Lisa               4
Nurse Betty             3
Homegrown               3
Head Above Water        3
Family Plot             3
Teaching Mrs. Tingle    2
Another Stakeout        2
Name: rating, dtype: int64

It counts how many time each movie was watched from 'short_title' column by using value_counts()

In [270]:
MyPartnerData.short_title.value_counts().head(10)

Nurse Betty             80
Mona Lisa               38
Zero Effect             34
Another Stakeout        29
Teaching Mrs. Tingle    22
Family Plot             20
Homegrown                9
Head Above Water         2
Name: short_title, dtype: int64

**My information**
    - Gender: Female
    - Genre preferences: Comedy
    - Age group: 18-25

'MyData' contains data for age group between (18-25), female gender, and genre is Comedy|Thriller(combination of both partner)

In [271]:
MyData = MergedData[(MergedData.age >= 18) & (MergedData.age <= 25) & (MergedData.gender == 'F') & (MergedData.genres == 'Comedy|Thriller')]
MyData.head()

user_id gender  age  occupation    zip  movie_id  rating   timestamp  \
180382      167      F   25          11  10022      3893       3   977178069   
180390      249      F   18          14  48126      3893       4   976728203   
180396      310      F   18           4  31207      3893       4  1012082324   
180404      361      F   25          14  94115      3893       4   976640712   
180408      410      F   25           1  55417      3893       3   976289449   

                     title           genres  year  short_title  
180382  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty  
180390  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty  
180396  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty  
180404  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty  
180408  Nurse Betty (2000)  Comedy|Thriller  2000  Nurse Betty

'MyGroupbyData' groups the 'short_title' column based on movie names by using groupby and finds the median of rating for each movie from 'rating' column by using median(). Then it sorts the rating column in decending order by using sort_values(ascending=False).

In [272]:
MyGroupbyData = MyData.groupby('short_title')['rating'].median()
MyGroupbyData.sort_values(ascending=False)

short_title
Zero Effect             4.0
Nurse Betty             4.0
Mona Lisa               3.0
Homegrown               3.0
Family Plot             3.0
Teaching Mrs. Tingle    2.5
Another Stakeout        2.0
Name: rating, dtype: float64

It counts how many time each movie was watched from 'short_title' column by using value_counts()

In [273]:
MyData.short_title.value_counts().head()

Nurse Betty             90
Zero Effect             36
Teaching Mrs. Tingle    32
Another Stakeout        23
Homegrown               17
Name: short_title, dtype: int64

**"Nurse Betty"** is our good rated movie for date night.

**Explanation:** Based on the analysis for my partner, 'Nurse Betty' was watched 80 times and rated 3/5 by his age group and gender. Based on the analysis for myself, 'Nurse Betty' was watched 90 times and rated 4/5 by my age group and gender. This movie is the most watched movie between me and my partner age group. Also, the genre of the movie meets our choice since my partner prefer thriller movie and I prefer comedy movie. Nurse Betty is the combination of both our prefer genre. Therefore, 'Nurse Betty' is a good movie for our date night.

## Part 2 (40 pts)

Load the dataset in `titanic.xls`. It contains data on all the passengers that travelled on the Titanic.

In [274]:
from IPython.core.display import HTML
HTML(filename='../data/titanic.html')

In [275]:
# you would need xlrd - pip install xlrd
t_file = pd.ExcelFile('../data/titanic.xls')
t_df = t_file.parse("titanic", header=None)
t_df.head()

0         1                                     2       3       4   \
0  pclass  survived                                  name     sex     age   
1       1         1         Allen, Miss. Elisabeth Walton  female      29   
2       1         1        Allison, Master. Hudson Trevor    male  0.9167   
3       1         0          Allison, Miss. Helen Loraine  female       2   
4       1         0  Allison, Mr. Hudson Joshua Creighton    male      30   

      5      6       7        8        9         10    11    12  \
0  sibsp  parch  ticket     fare    cabin  embarked  boat  body   
1      0      0   24160  211.338       B5         S     2   NaN   
2      1      2  113781   151.55  C22 C26         S    11   NaN   
3      1      2  113781   151.55  C22 C26         S   NaN   NaN   
4      1      2  113781   151.55  C22 C26         S   NaN   135   

                                13  
0                        home.dest  
1                     St Louis, MO  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

Rename t_df columns accordingly by using rename(). Then delete the first row (index 0).

In [276]:
t_df.rename(columns={0: 'pclass', 1: 'survived', 2: 'name', 3: 'sex', 4: 'age', 5: 'sibsp', 6: 'parch', 7: 'ticket', 8: 'fare', 9: 'cabin', 10: 'embarked', 11: 'boat', 12: 'body', 13: 'home.dest'}, inplace=True)
t_df = t_df.drop(0)
t_df.head()

pclass survived                                             name     sex  \
1      1        1                    Allen, Miss. Elisabeth Walton  female   
2      1        1                   Allison, Master. Hudson Trevor    male   
3      1        0                     Allison, Miss. Helen Loraine  female   
4      1        0             Allison, Mr. Hudson Joshua Creighton    male   
5      1        0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

      age sibsp parch  ticket     fare    cabin embarked boat body  \
1      29     0     0   24160  211.338       B5        S    2  NaN   
2  0.9167     1     2  113781   151.55  C22 C26        S   11  NaN   
3       2     1     2  113781   151.55  C22 C26        S  NaN  NaN   
4      30     1     2  113781   151.55  C22 C26        S  NaN  135   
5      25     1     2  113781   151.55  C22 C26        S  NaN  NaN   

                         home.dest  
1                     St Louis, MO  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON  
5  Montreal, PQ / Chesterville, ON

### Women and children first?

*** 1. Use the `groupby` method to calculate the proportion of passengers that survived by sex. (10 pts)*** 

'GroupbySurvival' will group the 't_df' based on 'survived" column by using groupby()

In [277]:
GroupbySurvival = t_df.groupby('survived')
GroupbySurvival.head()

pclass survived                                             name     sex  \
1       1        1                    Allen, Miss. Elisabeth Walton  female   
2       1        1                   Allison, Master. Hudson Trevor    male   
3       1        0                     Allison, Miss. Helen Loraine  female   
4       1        0             Allison, Mr. Hudson Joshua Creighton    male   
5       1        0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   
6       1        1                              Anderson, Mr. Harry    male   
7       1        1                Andrews, Miss. Kornelia Theodosia  female   
8       1        0                           Andrews, Mr. Thomas Jr    male   
9       1        1    Appleton, Mrs. Edward Dale (Charlotte Lamson)  female   
10      1        0                          Artagaveytia, Mr. Ramon    male   

       age sibsp parch    ticket     fare    cabin embarked boat body  \
1       29     0     0     24160  211.338       B5        S    2  NaN   
2   0.9167     1     2    113781   151.55  C22 C26        S   11  NaN   
3        2     1     2    113781   151.55  C22 C26        S  NaN  NaN   
4       30     1     2    113781   151.55  C22 C26        S  NaN  135   
5       25     1     2    113781   151.55  C22 C26        S  NaN  NaN   
6       48     0     0     19952    26.55      E12        S    3  NaN   
7       63     1     0     13502  77.9583       D7        S   10  NaN   
8       39     0     0    112050        0      A36        S  NaN  NaN   
9       53     2     0     11769  51.4792     C101        S    D  NaN   
10      71     0     0  PC 17609  49.5042      NaN        C  NaN   22   

                          home.dest  
1                      St Louis, MO  
2   Montreal, PQ / Chesterville, ON  
3   Montreal, PQ / Chesterville, ON  
4   Montreal, PQ / Chesterville, ON  
5   Montreal, PQ / Chesterville, ON  
6                      New York, NY  
7                        Hudson, NY  
8                       Belfast, NI  
9               Bayside, Queens, NY  
10              Montevideo, Uruguay

Calculating the total number of passenger who survived

In [278]:
TotalPassengerSurvived = GroupbySurvival.get_group(1).survived.count()
print(TotalPassengerSurvived)

500


It groups the survived people by gender based on 'sex' column by using groupby()

In [279]:
SurvivedByGender = GroupbySurvival.get_group(1).groupby('sex')
SurvivedByGender.head()

pclass survived                                               name     sex  \
1       1        1                      Allen, Miss. Elisabeth Walton  female   
2       1        1                     Allison, Master. Hudson Trevor    male   
6       1        1                                Anderson, Mr. Harry    male   
7       1        1                  Andrews, Miss. Kornelia Theodosia  female   
9       1        1      Appleton, Mrs. Edward Dale (Charlotte Lamson)  female   
12      1        1  Astor, Mrs. John Jacob (Madeleine Talmadge Force)  female   
13      1        1                      Aubart, Mme. Leontine Pauline  female   
15      1        1               Barkworth, Mr. Algernon Henry Wilson    male   
21      1        1                      Beckwith, Mr. Richard Leonard    male   
23      1        1                              Behr, Mr. Karl Howell    male   

       age sibsp parch    ticket     fare    cabin embarked boat body  \
1       29     0     0     24160  211.338       B5        S    2  NaN   
2   0.9167     1     2    113781   151.55  C22 C26        S   11  NaN   
6       48     0     0     19952    26.55      E12        S    3  NaN   
7       63     1     0     13502  77.9583       D7        S   10  NaN   
9       53     2     0     11769  51.4792     C101        S    D  NaN   
12      18     1     0  PC 17757  227.525  C62 C64        C    4  NaN   
13      24     0     0  PC 17477     69.3      B35        C    9  NaN   
15      80     0     0     27042       30      A23        S    B  NaN   
21      37     1     1     11751  52.5542      D35        S    5  NaN   
23      26     0     0    111369       30     C148        C    5  NaN   

                          home.dest  
1                      St Louis, MO  
2   Montreal, PQ / Chesterville, ON  
6                      New York, NY  
7                        Hudson, NY  
9               Bayside, Queens, NY  
12                     New York, NY  
13                    Paris, France  
15                    Hessle, Yorks  
21                     New York, NY  
23                     New York, NY

Calculating total number of female and male who survived

In [280]:
FemaleSurvived = SurvivedByGender.get_group('female').survived.count()
print("Female survived: " + str(FemaleSurvived))
MaleSurvived = SurvivedByGender.get_group('male').survived.count()
print("Male survived: " + str(MaleSurvived))

Female survived: 339
Male survived: 161


Calculating the proportion of passengers that survived by sex. <br>
Formula: <br>
sex_proportion = sex_survived / total_passenger_survived

In [281]:
FemaleSurvivedRatio = FemaleSurvived / TotalPassengerSurvived
print("Female proportion: " + str(FemaleSurvivedRatio))
MaleSurvivedRatio = MaleSurvived / TotalPassengerSurvived
print("Male proportion: " + str(MaleSurvivedRatio))

Female proportion: 0.678
Male proportion: 0.322


*** 2. Calculate the same proportion, but by class and sex. (10 pts) *** 

There are three passenger classes in the given dataframe. Creating three different dataframes by the passenger class.

In [282]:
PassengerClass1 = t_df.loc[t_df['pclass'] == 1]
PassengerClass2 = t_df.loc[t_df['pclass'] == 2]
PassengerClass3 = t_df.loc[t_df['pclass'] == 3]

In [283]:
PassengerClass1.head()

pclass survived                                             name     sex  \
1      1        1                    Allen, Miss. Elisabeth Walton  female   
2      1        1                   Allison, Master. Hudson Trevor    male   
3      1        0                     Allison, Miss. Helen Loraine  female   
4      1        0             Allison, Mr. Hudson Joshua Creighton    male   
5      1        0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

      age sibsp parch  ticket     fare    cabin embarked boat body  \
1      29     0     0   24160  211.338       B5        S    2  NaN   
2  0.9167     1     2  113781   151.55  C22 C26        S   11  NaN   
3       2     1     2  113781   151.55  C22 C26        S  NaN  NaN   
4      30     1     2  113781   151.55  C22 C26        S  NaN  135   
5      25     1     2  113781   151.55  C22 C26        S  NaN  NaN   

                         home.dest  
1                     St Louis, MO  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON  
5  Montreal, PQ / Chesterville, ON

In [284]:
PassengerClass2.head()

pclass survived                                   name     sex age sibsp  \
324      2        0                    Abelson, Mr. Samuel    male  30     1   
325      2        1  Abelson, Mrs. Samuel (Hannah Wizosky)  female  28     1   
326      2        0         Aldworth, Mr. Charles Augustus    male  30     0   
327      2        0             Andrew, Mr. Edgardo Samuel    male  18     0   
328      2        0               Andrew, Mr. Frank Thomas    male  25     0   

    parch      ticket  fare cabin embarked boat body  \
324     0   P/PP 3381    24   NaN        C  NaN  NaN   
325     0   P/PP 3381    24   NaN        C   10  NaN   
326     0      248744    13   NaN        S  NaN  NaN   
327     0      231945  11.5   NaN        S  NaN  NaN   
328     0  C.A. 34050  10.5   NaN        S  NaN  NaN   

                                    home.dest  
324                       Russia New York, NY  
325                       Russia New York, NY  
326                        Bryn Mawr, PA, USA  
327  Buenos Aires, Argentina / New Jersey, NJ  
328            Cornwall, England Houghton, MI

In [285]:
PassengerClass3.head()

pclass survived                              name     sex age sibsp parch  \
601      3        0               Abbing, Mr. Anthony    male  42     0     0   
602      3        0     Abbott, Master. Eugene Joseph    male  13     0     2   
603      3        0       Abbott, Mr. Rossmore Edward    male  16     1     1   
604      3        1  Abbott, Mrs. Stanton (Rosa Hunt)  female  35     1     1   
605      3        1       Abelseth, Miss. Karen Marie  female  16     0     0   

        ticket   fare cabin embarked boat body               home.dest  
601  C.A. 5547   7.55   NaN        S  NaN  NaN                     NaN  
602  C.A. 2673  20.25   NaN        S  NaN  NaN     East Providence, RI  
603  C.A. 2673  20.25   NaN        S  NaN  190     East Providence, RI  
604  C.A. 2673  20.25   NaN        S    A  NaN     East Providence, RI  
605     348125   7.65   NaN        S   16  NaN  Norway Los Angeles, CA

Creating seperate dataframes by the passenger class and gender

In [286]:
# Male and female in class 1
MalePassengerClass1 = PassengerClass1[PassengerClass1['sex'] == 'male']
FemalePassengerClass1 = PassengerClass1[PassengerClass1['sex'] == 'female']
# Male and female in class 2
MalePassengerClass2 = PassengerClass2[PassengerClass2['sex'] == 'male']
FemalePassengerClass2 = PassengerClass2[PassengerClass2['sex'] == 'female']
# Male and female in class 3
MalePassengerClass3 = PassengerClass3[PassengerClass3['sex'] == 'male']
FemalePassengerClass3 = PassengerClass3[PassengerClass3['sex'] == 'female']

In [287]:
FemalePassengerClass1.head()

pclass survived                                             name     sex  \
1      1        1                    Allen, Miss. Elisabeth Walton  female   
3      1        0                     Allison, Miss. Helen Loraine  female   
5      1        0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   
7      1        1                Andrews, Miss. Kornelia Theodosia  female   
9      1        1    Appleton, Mrs. Edward Dale (Charlotte Lamson)  female   

  age sibsp parch  ticket     fare    cabin embarked boat body  \
1  29     0     0   24160  211.338       B5        S    2  NaN   
3   2     1     2  113781   151.55  C22 C26        S  NaN  NaN   
5  25     1     2  113781   151.55  C22 C26        S  NaN  NaN   
7  63     1     0   13502  77.9583       D7        S   10  NaN   
9  53     2     0   11769  51.4792     C101        S    D  NaN   

                         home.dest  
1                     St Louis, MO  
3  Montreal, PQ / Chesterville, ON  
5  Montreal, PQ / Chesterville, ON  
7                       Hudson, NY  
9              Bayside, Queens, NY

In [288]:
MalePassengerClass3.head()

pclass survived                                      name   sex age sibsp  \
601      3        0                       Abbing, Mr. Anthony  male  42     0   
602      3        0             Abbott, Master. Eugene Joseph  male  13     0   
603      3        0               Abbott, Mr. Rossmore Edward  male  16     1   
606      3        1             Abelseth, Mr. Olaus Jorgensen  male  25     0   
607      3        1  Abrahamsson, Mr. Abraham August Johannes  male  20     0   

    parch            ticket   fare  cabin embarked boat body  \
601     0         C.A. 5547   7.55    NaN        S  NaN  NaN   
602     2         C.A. 2673  20.25    NaN        S  NaN  NaN   
603     1         C.A. 2673  20.25    NaN        S  NaN  190   
606     0            348122   7.65  F G63        S    A  NaN   
607     0  SOTON/O2 3101284  7.925    NaN        S   15  NaN   

                             home.dest  
601                                NaN  
602                East Providence, RI  
603                East Providence, RI  
606                 Perkins County, SD  
607  Taalintehdas, Finland Hoboken, NJ

Calculating the total number of people who survived in each passenger class and gender

In [289]:
# Male and female survived in class 1
MalePassengerClass1Survived = MalePassengerClass1[MalePassengerClass1['survived'] == 1].survived.count()
print("Male class-1 survived: " + str(MalePassengerClass1Survived))
FemalePassengerClass1Survived = FemalePassengerClass1[FemalePassengerClass1['survived'] == 1].survived.count()
print("Female class-1 survived: " + str(FemalePassengerClass1Survived) + '\n')

# Male and female survived in class 2
MalePassengerClass2Survived = MalePassengerClass2[MalePassengerClass2['survived'] == 1].survived.count()
print("Male class-2 survived: " + str(MalePassengerClass2Survived))
FemalePassengerClass2Survived = FemalePassengerClass2[FemalePassengerClass2['survived'] == 1].survived.count()
print("Female class-2 survived: " + str(FemalePassengerClass2Survived) + '\n')

# Male and female survived in class 3
MalePassengerClass3Survived = MalePassengerClass3[MalePassengerClass3['survived'] == 1].survived.count()
print("Male class-3 survived: " + str(MalePassengerClass3Survived))
FemalePassengerClass3Survived = FemalePassengerClass3[FemalePassengerClass3['survived'] == 1].survived.count()
print("Female class-3 survived: " + str(FemalePassengerClass3Survived))

Male class-1 survived: 61
Female class-1 survived: 139

Male class-2 survived: 25
Female class-2 survived: 94

Male class-3 survived: 75
Female class-3 survived: 106


Calculating the proportion of passengers that survived by class and sex. <br>
Formula: <br>
gender_class_ratio = gender_class_survived / total_passenger_survived

In [290]:
# Male and female survived ratio in class 1
MaleClass1Ratio = MalePassengerClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Male Ratio: " + str(MaleClass1Ratio))
FemaleClass1Ratio = FemalePassengerClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Female Ratio: " + str(FemaleClass1Ratio) + '\n')

# Male and female survived ratio in class 2
MaleClass2Ratio = MalePassengerClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Male Ratio: " + str(MaleClass2Ratio))
FemaleClass2Ratio = FemalePassengerClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Female Ratio: " + str(FemaleClass2Ratio) + '\n')

# Male and female survived ratio in class 3
MaleClass3Ratio = MalePassengerClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Male Ratio: " + str(MaleClass3Ratio))
FemaleClass3Ratio = FemalePassengerClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Female Ratio: " + str(FemaleClass3Ratio))

Passenger Class-1 Male Ratio: 0.122
Passenger Class-1 Female Ratio: 0.278

Passenger Class-2 Male Ratio: 0.05
Passenger Class-2 Female Ratio: 0.188

Passenger Class-3 Male Ratio: 0.15
Passenger Class-3 Female Ratio: 0.212


*** 3. Create age categories: children (under 14 years), adolescents (14-20), adult (21-64), and senior(65+), and calculate survival proportions by age category, class and sex. (20 pts)***

Creating a new column name 'age_category' and iterate through the dataframe to fill the column properly

In [291]:
t_df["age_category"] = ""

for x in range(1, len(t_df)+1):
    if (t_df.age[x] < 14):
        t_df.age_category[x] = "children"
    elif (t_df.age[x] >= 14) and (t_df.age[x] < 21):
        t_df.age_category[x] = "adolescents"
    elif (t_df.age[x] >= 21) and (t_df.age[x] < 65):
        t_df.age_category[x] = "adult"
    elif (t_df.age[x] >= 65):
        t_df.age_category[x] = "senior"
    else:
        t_df.age_category[x] = "NaN"

Creating seperate dataframes based on the age by using 'age-category' column

In [292]:
children = t_df.loc[t_df['age_category'] == "children"]
adolescents = t_df.loc[t_df['age_category'] == "adolescents"]
adult = t_df.loc[t_df['age_category'] == "adult"]
senior = t_df.loc[t_df['age_category'] == "senior"]

In [293]:
adolescents.head()

pclass survived                                               name     sex  \
12      1        1  Astor, Mrs. John Jacob (Madeleine Talmadge Force)  female   
28      1        1            Bishop, Mrs. Dickinson H (Helen Walton)  female   
54      1        0                             Carrau, Mr. Jose Pedro    male   
56      1        1                          Carter, Miss. Lucile Polk  female   
93      1        1          Dick, Mrs. Albert Adrian (Vera Gillespie)  female   

   age sibsp parch    ticket     fare    cabin embarked boat body  \
12  18     1     0  PC 17757  227.525  C62 C64        C    4  NaN   
28  19     1     0     11967  91.0792      B49        C    7  NaN   
54  17     0     0    113059     47.1      NaN        S  NaN  NaN   
56  14     1     2    113760      120  B96 B98        S    4  NaN   
93  17     1     0     17474       57      B20        S    3  NaN   

              home.dest age_category  
12         New York, NY  adolescents  
28         Dowagiac, MI  adolescents  
54  Montevideo, Uruguay  adolescents  
56        Bryn Mawr, PA  adolescents  
93          Calgary, AB  adolescents

In [294]:
senior.head()

pclass survived                                               name  \
10       1        0                            Artagaveytia, Mr. Ramon   
15       1        1               Barkworth, Mr. Algernon Henry Wilson   
62       1        1  Cavendish, Mrs. Tyrell William (Julia Florence...   
82       1        0                       Crosby, Capt. Edward Gifford   
136      1        0                          Goldschmidt, Mr. George B   

        sex age sibsp parch     ticket     fare cabin embarked boat body  \
10     male  71     0     0   PC 17609  49.5042   NaN        C  NaN   22   
15     male  80     0     0      27042       30   A23        S    B  NaN   
62   female  76     1     0      19877    78.85   C46        S    6  NaN   
82     male  70     1     1  WE/P 5735       71   B22        S  NaN  269   
136    male  71     0     0   PC 17754  34.6542    A5        C  NaN  NaN   

                   home.dest age_category  
10       Montevideo, Uruguay       senior  
15             Hessle, Yorks       senior  
62   Little Onn Hall, Staffs       senior  
82             Milwaukee, WI       senior  
136             New York, NY       senior

Createing seperate dataframes based on the age_category and sex

In [295]:
# Male and female children
MaleChildren = children.loc[children['sex'] == 'male']
FemaleChildren = children.loc[children['sex'] == 'female']

# Male and female adolescents
MaleAdolescents = adolescents.loc[adolescents['sex'] == 'male']
FemaleAdolescents = adolescents.loc[adolescents['sex'] == 'female']

# Male and female adult
MaleAdult = adult.loc[adult['sex'] == 'male']
FemaleAdult = adult.loc[adult['sex'] == 'female']

# Male and female senior
MaleSenior = senior.loc[senior['sex'] == 'male']
FemaleSenior = senior.loc[senior['sex'] == 'female']

In [296]:
FemaleChildren.head()

pclass survived                               name     sex age sibsp  \
3        1        0       Allison, Miss. Helen Loraine  female   2     1   
341      2        1        Becker, Miss. Marion Louise  female   4     2   
342      2        1       Becker, Miss. Ruth Elizabeth  female  12     2   
378      2        1   Collyer, Miss. Marjorie "Lottie"  female   8     0   
431      2        1  Harper, Miss. Annie Jessie "Nina"  female   6     0   

    parch      ticket    fare    cabin embarked boat body  \
3       2      113781  151.55  C22 C26        S  NaN  NaN   
341     1      230136      39       F4        S   11  NaN   
342     1      230136      39       F4        S   13  NaN   
378     2  C.A. 31921   26.25      NaN        S   14  NaN   
431     1      248727      33      NaN        S   11  NaN   

                                   home.dest age_category  
3            Montreal, PQ / Chesterville, ON     children  
341       Guntur, India / Benton Harbour, MI     children  
342       Guntur, India / Benton Harbour, MI     children  
378  Bishopstoke, Hants / Fayette Valley, ID     children  
431           Denmark Hill, Surrey / Chicago     children

In [297]:
MaleAdult.head()

pclass survived                                  name   sex age sibsp  \
4       1        0  Allison, Mr. Hudson Joshua Creighton  male  30     1   
6       1        1                   Anderson, Mr. Harry  male  48     0   
8       1        0                Andrews, Mr. Thomas Jr  male  39     0   
11      1        0                Astor, Col. John Jacob  male  47     1   
17      1        0              Baxter, Mr. Quigg Edmond  male  24     0   

   parch    ticket     fare    cabin embarked boat body  \
4      2    113781   151.55  C22 C26        S  NaN  135   
6      0     19952    26.55      E12        S    3  NaN   
8      0    112050        0      A36        S  NaN  NaN   
11     0  PC 17757  227.525  C62 C64        C  NaN  124   
17     1  PC 17558  247.521  B58 B60        C  NaN  NaN   

                          home.dest age_category  
4   Montreal, PQ / Chesterville, ON        adult  
6                      New York, NY        adult  
8                       Belfast, NI        adult  
11                     New York, NY        adult  
17                     Montreal, PQ        adult

In [298]:
FemaleSenior.head()

pclass survived                                               name     sex  \
62      1        1  Cavendish, Mrs. Tyrell William (Julia Florence...  female   

   age sibsp parch ticket   fare cabin embarked boat body  \
62  76     1     0  19877  78.85   C46        S    6  NaN   

                  home.dest age_category  
62  Little Onn Hall, Staffs       senior

Creating seperate dataframes based on the age_category, sex, and class

In [299]:
# Male and female children in class 1,2,3
MaleChildrenClass1 = MaleChildren.loc[MaleChildren['pclass'] == 1]
MaleChildrenClass2 = MaleChildren.loc[MaleChildren['pclass'] == 2]
MaleChildrenClass3 = MaleChildren.loc[MaleChildren['pclass'] == 3]
FemaleChildrenClass1 = FemaleChildren.loc[FemaleChildren['pclass'] == 1]
FemaleChildrenClass2 = FemaleChildren.loc[FemaleChildren['pclass'] == 2]
FemaleChildrenClass3 = FemaleChildren.loc[FemaleChildren['pclass'] == 3]

# Male and female adolescents in class 1,2,3
MaleAdolescentsClass1 = MaleAdolescents.loc[MaleAdolescents['pclass'] == 1]
MaleAdolescentsClass2 = MaleAdolescents.loc[MaleAdolescents['pclass'] == 2]
MaleAdolescentsClass3 = MaleAdolescents.loc[MaleAdolescents['pclass'] == 3]
FemaleAdolescentsClass1 = FemaleAdolescents.loc[FemaleAdolescents['pclass'] == 1]
FemaleAdolescentsClass2 = FemaleAdolescents.loc[FemaleAdolescents['pclass'] == 2]
FemaleAdolescentsClass3 = FemaleAdolescents.loc[FemaleAdolescents['pclass'] == 3]

# Male and female adult in class 1,2,3
MaleAdultClass1 = MaleAdult.loc[MaleAdult['pclass'] == 1]
MaleAdultClass2 = MaleAdult.loc[MaleAdult['pclass'] == 2]
MaleAdultClass3 = MaleAdult.loc[MaleAdult['pclass'] == 3]
FemaleAdultClass1 = FemaleAdult.loc[FemaleAdult['pclass'] == 1]
FemaleAdultClass2 = FemaleAdult.loc[FemaleAdult['pclass'] == 2]
FemaleAdultClass3 = FemaleAdult.loc[FemaleAdult['pclass'] == 3]

# Male and female senior in class 1,2,3
MaleSeniorClass1 = MaleSenior.loc[MaleSenior['pclass'] == 1]
MaleSeniorClass2 = MaleSenior.loc[MaleSenior['pclass'] == 2]
MaleSeniorClass3 = MaleSenior.loc[MaleSenior['pclass'] == 3]
FemaleSeniorClass1 = FemaleSenior.loc[FemaleSenior['pclass'] == 1]
FemaleSeniorClass2 = FemaleSenior.loc[FemaleSenior['pclass'] == 2]
FemaleSeniorClass3 = FemaleSenior.loc[FemaleSenior['pclass'] == 3]

In [300]:
FemaleChildrenClass1.head()

pclass survived                          name     sex age sibsp parch  \
3      1        0  Allison, Miss. Helen Loraine  female   2     1     2   

   ticket    fare    cabin embarked boat body  \
3  113781  151.55  C22 C26        S  NaN  NaN   

                         home.dest age_category  
3  Montreal, PQ / Chesterville, ON     children

In [301]:
FemaleAdolescentsClass1.head()

pclass survived                                               name  \
12       1        1  Astor, Mrs. John Jacob (Madeleine Talmadge Force)   
28       1        1            Bishop, Mrs. Dickinson H (Helen Walton)   
56       1        1                          Carter, Miss. Lucile Polk   
93       1        1          Dick, Mrs. Albert Adrian (Vera Gillespie)   
138      1        1                       Graham, Miss. Margaret Edith   

        sex age sibsp parch    ticket     fare    cabin embarked boat body  \
12   female  18     1     0  PC 17757  227.525  C62 C64        C    4  NaN   
28   female  19     1     0     11967  91.0792      B49        C    7  NaN   
56   female  14     1     2    113760      120  B96 B98        S    4  NaN   
93   female  17     1     0     17474       57      B20        S    3  NaN   
138  female  19     0     0    112053       30      B42        S    3  NaN   

         home.dest age_category  
12    New York, NY  adolescents  
28    Dowagiac, MI  adolescents  
56   Bryn Mawr, PA  adolescents  
93     Calgary, AB  adolescents  
138  Greenwich, CT  adolescents

In [302]:
MaleAdultClass2.head()

pclass survived                            name   sex age sibsp parch  \
324      2        0             Abelson, Mr. Samuel  male  30     1     0   
326      2        0  Aldworth, Mr. Charles Augustus  male  30     0     0   
328      2        0        Andrew, Mr. Frank Thomas  male  25     0     0   
329      2        0            Angle, Mr. William A  male  34     1     0   
331      2        0                 Ashby, Mr. John  male  57     0     0   

         ticket  fare cabin embarked boat body  \
324   P/PP 3381    24   NaN        C  NaN  NaN   
326      248744    13   NaN        S  NaN  NaN   
328  C.A. 34050  10.5   NaN        S  NaN  NaN   
329      226875    26   NaN        S  NaN  NaN   
331      244346    13   NaN        S  NaN  NaN   

                          home.dest age_category  
324             Russia New York, NY        adult  
326              Bryn Mawr, PA, USA        adult  
328  Cornwall, England Houghton, MI        adult  
329                Warwick, England        adult  
331                West Hoboken, NJ        adult

In [303]:
MaleSeniorClass1.head()

pclass survived                                  name   sex age sibsp  \
10       1        0               Artagaveytia, Mr. Ramon  male  71     0   
15       1        1  Barkworth, Mr. Algernon Henry Wilson  male  80     0   
82       1        0          Crosby, Capt. Edward Gifford  male  70     1   
136      1        0             Goldschmidt, Mr. George B  male  71     0   
206      1        0             Millet, Mr. Francis Davis  male  65     0   

    parch     ticket     fare cabin embarked boat body             home.dest  \
10      0   PC 17609  49.5042   NaN        C  NaN   22   Montevideo, Uruguay   
15      0      27042       30   A23        S    B  NaN         Hessle, Yorks   
82      1  WE/P 5735       71   B22        S  NaN  269         Milwaukee, WI   
136     0   PC 17754  34.6542    A5        C  NaN  NaN          New York, NY   
206     0      13509    26.55   E38        S  NaN  249  East Bridgewater, MA   

    age_category  
10        senior  
15        senior  
82        senior  
136       senior  
206       senior

Calculating total number of male and female who survived in each age_category and passenger class

In [304]:
# Male and female children survived in class 1,2,3
MaleChildrenClass1Survived = MaleChildrenClass1[MaleChildrenClass1['survived'] == 1].survived.count()
print("Male children class-1 survived: " + str(MaleChildrenClass1Survived))
MaleChildrenClass2Survived = MaleChildrenClass2[MaleChildrenClass2['survived'] == 1].survived.count()
print("Male children class-2 survived: " + str(MaleChildrenClass2Survived))
MaleChildrenClass3Survived = MaleChildrenClass3[MaleChildrenClass3['survived'] == 1].survived.count()
print("Male children class-3 survived: " + str(MaleChildrenClass3Survived))
FemaleChildrenClass1Survived = FemaleChildrenClass1[FemaleChildrenClass1['survived'] == 1].survived.count()
print("Female children class-1 survived: " + str(FemaleChildrenClass1Survived))
FemaleChildrenClass2Survived = FemaleChildrenClass2[FemaleChildrenClass2['survived'] == 1].survived.count()
print("Female children class-2 survived: " + str(FemaleChildrenClass2Survived))
FemaleChildrenClass3Survived = FemaleChildrenClass3[FemaleChildrenClass3['survived'] == 1].survived.count()
print("Female children class-3 survived: " + str(FemaleChildrenClass3Survived) + '\n')

# Male and female adolescents survived in class 1,2,3
MaleAdolescentsClass1Survived = MaleAdolescentsClass1[MaleAdolescentsClass1['survived'] == 1].survived.count()
print("Male adolescents class-1 survived: " + str(MaleAdolescentsClass1Survived))
MaleAdolescentsClass2Survived = MaleAdolescentsClass2[MaleAdolescentsClass2['survived'] == 1].survived.count()
print("Male adolescents class-2 survived: " + str(MaleAdolescentsClass2Survived))
MaleAdolescentsClass3Survived = MaleAdolescentsClass3[MaleAdolescentsClass3['survived'] == 1].survived.count()
print("Male adolescents class-3 survived: " + str(MaleAdolescentsClass3Survived))
FemaleAdolescentsClass1Survived = FemaleAdolescentsClass1[FemaleAdolescentsClass1['survived'] == 1].survived.count()
print("Female adolescents class-1 survived: " + str(FemaleAdolescentsClass1Survived))
FemaleAdolescentsClass2Survived = FemaleAdolescentsClass2[FemaleAdolescentsClass2['survived'] == 1].survived.count()
print("Female adolescents class-2 survived: " + str(FemaleAdolescentsClass2Survived))
FemaleAdolescentsClass3Survived = FemaleAdolescentsClass3[FemaleAdolescentsClass3['survived'] == 1].survived.count()
print("Female adolescents class-3 survived: " + str(FemaleAdolescentsClass3Survived) + '\n')

# Male and female adult survived in class 1,2,3
MaleAdultClass1Survived = MaleAdultClass1[MaleAdultClass1['survived'] == 1].survived.count()
print("Male adult class-1 survived: " + str(MaleAdultClass1Survived))
MaleAdultClass2Survived = MaleAdultClass2[MaleAdultClass2['survived'] == 1].survived.count()
print("Male adult class-2 survived: " + str(MaleAdultClass2Survived))
MaleAdultClass3Survived = MaleAdultClass3[MaleAdultClass3['survived'] == 1].survived.count()
print("Male adult class-3 survived: " + str(MaleAdultClass3Survived))
FemaleAdultClass1Survived = FemaleAdultClass1[FemaleAdultClass1['survived'] == 1].survived.count()
print("Female adult class-1 survived: " + str(FemaleAdultClass1Survived))
FemaleAdultClass2Survived = FemaleAdultClass2[FemaleAdultClass2['survived'] == 1].survived.count()
print("Female adult class-2 survived: " + str(FemaleAdultClass2Survived))
FemaleAdultClass3Survived = FemaleAdultClass3[FemaleAdultClass3['survived'] == 1].survived.count()
print("Female adult class-3 survived: " + str(FemaleAdultClass3Survived) + '\n')

# Male and female senior survived in class 1,2,3
MaleSeniorClass1Survived = MaleSeniorClass1[MaleSeniorClass1['survived'] == 1].survived.count()
print("Male senior class-1 survived: " + str(MaleSeniorClass1Survived))
MaleSeniorClass2Survived = MaleSeniorClass2[MaleSeniorClass2['survived'] == 1].survived.count()
print("Male senior class-2 survived: " + str(MaleSeniorClass2Survived))
MaleSeniorClass3Survived = MaleSeniorClass3[MaleSeniorClass3['survived'] == 1].survived.count()
print("Male senior class-3 survived: " + str(MaleSeniorClass3Survived))
FemaleSeniorClass1Survived = FemaleSeniorClass1[FemaleSeniorClass1['survived'] == 1].survived.count()
print("Female senior class-1 survived: " + str(FemaleSeniorClass1Survived))
FemaleSeniorClass2Survived = FemaleSeniorClass2[FemaleSeniorClass2['survived'] == 1].survived.count()
print("Female senior class-2 survived: " + str(FemaleSeniorClass2Survived))
FemaleSeniorClass3Survived = FemaleSeniorClass3[FemaleSeniorClass3['survived'] == 1].survived.count()
print("Female senior class-3 survived: " + str(FemaleSeniorClass3Survived))

Male children class-1 survived: 5
Male children class-2 survived: 11
Male children class-3 survived: 12
Female children class-1 survived: 0
Female children class-2 survived: 14
Female children class-3 survived: 15

Male adolescents class-1 survived: 1
Male adolescents class-2 survived: 2
Male adolescents class-3 survived: 8
Female adolescents class-1 survived: 15
Female adolescents class-2 survived: 12
Female adolescents class-3 survived: 19

Male adult class-1 survived: 46
Male adult class-2 survived: 10
Male adult class-3 survived: 39
Female adult class-1 survived: 112
Female adult class-2 survived: 66
Female adult class-3 survived: 38

Male senior class-1 survived: 1
Male senior class-2 survived: 0
Male senior class-3 survived: 0
Female senior class-1 survived: 1
Female senior class-2 survived: 0
Female senior class-3 survived: 0


Calculating survival proportions by age_category, class and sex. <br>
Formula: <br>
gender_age_class_ratio = gender_age_class_survived / total_passenger_survived

In [305]:
# Male and female children survival ratio in class 1,2,3
MaleChildrenClass1Ratio = MaleChildrenClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Male Children Ratio: " + str(MaleChildrenClass1Ratio))
MaleChildrenClass2Ratio = MaleChildrenClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Male Children Ratio: " + str(MaleChildrenClass2Ratio))
MaleChildrenClass3Ratio = MaleChildrenClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Male Children Ratio: " + str(MaleChildrenClass3Ratio))
FemaleChildrenClass1Ratio = FemaleChildrenClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Female Children Ratio: " + str(FemaleChildrenClass1Ratio))
FemaleChildrenClass2Ratio = FemaleChildrenClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Female Children Ratio: " + str(FemaleChildrenClass2Ratio))
FemaleChildrenClass3Ratio = FemaleChildrenClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Female Children Ratio: " + str(FemaleChildrenClass3Ratio) + '\n')

# Male and female adolescents survival ratio in class 1,2,3
MaleAdolescentsClass1Ratio = MaleAdolescentsClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Male Adolescents Ratio: " + str(MaleAdolescentsClass1Ratio))
MaleAdolescentsClass2Ratio = MaleAdolescentsClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Male Adolescents Ratio: " + str(MaleAdolescentsClass2Ratio))
MaleAdolescentsClass3Ratio = MaleAdolescentsClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Male Adolescents Ratio: " + str(MaleAdolescentsClass3Ratio))
FemaleAdolescentsClass1Ratio = FemaleAdolescentsClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Female Adolescents Ratio: " + str(FemaleAdolescentsClass1Ratio))
FemaleAdolescentsClass2Ratio = FemaleAdolescentsClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Female Adolescents Ratio: " + str(FemaleAdolescentsClass2Ratio))
FemaleAdolescentsClass3Ratio = FemaleAdolescentsClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Female Adolescents Ratio: " + str(FemaleAdolescentsClass3Ratio) + '\n')

# Male and female adult survival ratio in class 1,2,3
MaleAdultClass1Ratio = MaleAdultClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Male Adult Ratio: " + str(MaleAdultClass1Ratio))
MaleAdultClass2Ratio = MaleAdultClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Male Adult Ratio: " + str(MaleAdultClass2Ratio))
MaleAdultClass3Ratio = MaleAdultClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Male Adult Ratio: " + str(MaleAdultClass3Ratio))
FemaleAdultClass1Ratio = FemaleAdultClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Female Adult Ratio: " + str(FemaleAdultClass1Ratio))
FemaleAdultClass2Ratio = FemaleAdultClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Female Adult Ratio: " + str(FemaleAdultClass2Ratio))
FemaleAdultClass3Ratio = FemaleAdultClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Female Adult Ratio: " + str(FemaleAdultClass3Ratio) + '\n')

# Male and female senior survival ratio in class 1,2,3
MaleSeniorClass1Ratio = MaleSeniorClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Male Senior Ratio: " + str(MaleSeniorClass1Ratio))
MaleSeniorClass2Ratio = MaleSeniorClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Male Senior Ratio: " + str(MaleSeniorClass2Ratio))
MaleSeniorClass3Ratio = MaleSeniorClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Male Senior Ratio: " + str(MaleSeniorClass3Ratio))
FemaleSeniorClass1Ratio = FemaleSeniorClass1Survived / TotalPassengerSurvived
print("Passenger Class-1 Female Senior Ratio: " + str(FemaleSeniorClass1Ratio))
FemaleSeniorClass2Ratio = FemaleSeniorClass2Survived / TotalPassengerSurvived
print("Passenger Class-2 Female Senior Ratio: " + str(FemaleSeniorClass2Ratio))
FemaleSeniorClass3Ratio = FemaleSeniorClass3Survived / TotalPassengerSurvived
print("Passenger Class-3 Female Senior Ratio: " + str(FemaleSeniorClass3Ratio))

Passenger Class-1 Male Children Ratio: 0.01
Passenger Class-2 Male Children Ratio: 0.022
Passenger Class-3 Male Children Ratio: 0.024
Passenger Class-1 Female Children Ratio: 0.0
Passenger Class-2 Female Children Ratio: 0.028
Passenger Class-3 Female Children Ratio: 0.03

Passenger Class-1 Male Adolescents Ratio: 0.002
Passenger Class-2 Male Adolescents Ratio: 0.004
Passenger Class-3 Male Adolescents Ratio: 0.016
Passenger Class-1 Female Adolescents Ratio: 0.03
Passenger Class-2 Female Adolescents Ratio: 0.024
Passenger Class-3 Female Adolescents Ratio: 0.038

Passenger Class-1 Male Adult Ratio: 0.092
Passenger Class-2 Male Adult Ratio: 0.02
Passenger Class-3 Male Adult Ratio: 0.078
Passenger Class-1 Female Adult Ratio: 0.224
Passenger Class-2 Female Adult Ratio: 0.132
Passenger Class-3 Female Adult Ratio: 0.076

Passenger Class-1 Male Senior Ratio: 0.002
Passenger Class-2 Male Senior Ratio: 0.0
Passenger Class-3 Male Senior Ratio: 0.0
Passenger Class-1 Female Senior Ratio: 0.002
Passe